### DISCUSSION TITLE
Twilio Webhook not working on Vercel amongst deployment or production

### DISCUSSION THREAD

**Discussion by EmilioJD** [vercel/community#3704] [Help]
### Summary

I have a Twilio webhook that accepts a message whenever it comes in. It is routed to a POST HTTP request that then uploads the text contents to a Supabase. It works locally with `vercel dev` but seems to break on production and receives an undefined body. Giving me the error: Cannot read properties of undefined (reading 'Body')

This has been bugging me for about a day now and I can't figure out why it works locally but breaks upon deployment

```
const express = require("express");
const { MessagingResponse } = require("twilio").twiml;
const { createClient } = require("@supabase/supabase-js");
require("dotenv").config();

const app = express();

app.post("/api/track-sms", async (req, res) => {
  const text = await req.body.Body;
  console.log(text)
  const twiml = new MessagingResponse();

  const supabase = createClient(
    process.env.SUPAURL,
    process.env.SUPAKEY
  );
  

  try {
    const { data, error } = await supabase
      .from("table")
      .insert([{ title: "A text message", content: text }])
      .select();

    if (error) {
      throw error;
    }

    res.writeHead(200, { "Content-Type": "text/xml" });
    res.end(twiml.toString());
  } catch (e) {
    console.error(e);
    res.status(500).send("Internal Server Error");
  }
});

module.exports = app;
```

### Steps to Reproduce

Try out the code snippet

**Comment by amyegan** [comment]
Hey @EmilioJD. Try using the build in body parser as described in https://stackoverflow.com/a/51090698

I added `app.use(express.json());` just after `const app = express();` to the code snippet to get the body data in my deployed test site.

**Comment by EmilioJD** [comment]
Thanks for the reply @amyegan, on deployment I still seem to get an undefined value for 

```
  const text = await req.body.Body;
  console.log(text)
 ``` 
 
 even though it works locally with `vercel dev`. Not sure if I'm understanding Vercel serverless incorrectly.

**Reply by amyegan** [reply]
Do you get the body value if you log it like this instead?
```js
const body = req.body;
console.log("req.body", JSON.stringify(body).substring(0,200));
```
